sentry-kotlin-multiplatform-gradle-plugin/src/main/java/io/sentry/kotlin/multiplatform/gradle/FrameworkLinker.kt

```diff
@@ -0,0 +1,59 @@
+package io.sentry.kotlin.multiplatform.gradle
+
+import org.gradle.api.logging.Logger
+import org.jetbrains.kotlin.gradle.dsl.KotlinNativeBinaryContainer
+import org.jetbrains.kotlin.gradle.plugin.mpp.Framework
+import org.jetbrains.kotlin.gradle.plugin.mpp.TestExecutable
+
+/**
+ * Responsible for executing the linking.
+ * This involves configuring and linking binaries to the Sentry Cocoa framework.
+ */
+class FrameworkLinker(
+    private val logger: Logger
+) {
+    fun configureBinaries(
+        binaries: KotlinNativeBinaryContainer,
+        dynamicPath: String?,
+        staticPath: String?
+    ) {
+        binaries.all { binary ->
+            when (binary) {
+                is TestExecutable -> linkTestBinary(binary, chooseTestPath(dynamicPath, staticPath))
+                is Framework -> linkFrameworkBinary(binary, dynamicPath, staticPath)
+                else -> logger.info("Ignoring binary type: ${binary::class.java.simpleName}")
+            }
+        }
+    }
+
+    private fun chooseTestPath(dynamic: String?, static: String?) = when {
+        dynamic != null && static != null -> {
```

Comment:
do you really need this condition? could as well just omit it, and it'd always pick `dynamic` when it's available or fall back to static otherwise